cadc-log/src/main/java/ca/nrc/cadc/log/LogControlServlet.java

```diff
@@ -359,21 +366,147 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     /**
      * Check for proper group membership.
      */
-    private void authorize(HttpServletRequest request, boolean readOnly) throws AccessControlException, TransientException {
-        if (authorizerClassName != null && groupAuthorizer == null) {
-            throw new RuntimeException("CONFIG: group authorizer was configured but failed to load: " + authorizerClassName);
+    private void authorize(HttpServletRequest request, boolean readOnly)
+        throws AccessControlException, TransientException {
+
+        // Track exceptions thrown
+        AccessControlException acException = null;
+
+        // first check if request user matches authorized config file users
+        Set<Principal> authorizedUsers = getAuthorizedUserPrincipals();
```

Comment:
The config file is read twice instead of once. It should be read once and used for the remainder of the request. You just need to re-org use of getLogControlProperties (called once, read once, Properties object passed to other methods).

Reply:
Updated to read the properties file once per request.